src/main/java/org/spongepowered/api/command/CommandSource.java

```diff
@@ -36,13 +38,14 @@
  * <p>Examples of potential implementations include players, the server console,
  * Rcon clients, web-based clients, command blocks, and so on.</p>
  */
-public interface CommandSource extends MessageReceiver, Subject {
+public interface CommandSource extends ContextViewer, MessageReceiver, NamedContextual, Subject {
```

Comment:
I don't think that CommandSource itself needs to be a context viewer, its only valid for players really.